lib/client.js

```diff
@@ -369,8 +381,9 @@ ServiceClient.Response = class ServiceClientResponse {
 };
 
 ServiceClient.Error = class ServiceClientError extends Error {
-    constructor(originalError, type, response) {
-        super(`${type}. ${originalError.message || ''}`);
+    // eslint-disable-next-line max-params
+    constructor(name, originalError, type, response) {
```

Comment:
This is a breaking change of public API. Can we put name without changing the existing arguments?

Reply:
Is the `ServiceClient.Error`"public API"? I considered it private API, assuming that users of this library create their own errors. But maybe that assumption is wrong?

Reply:
Mmm, at least it's exposed as `ServiceClient.Error` and my team is using it in a kind of filter. But we can change it though. @grassator What do you think?

Reply:
As it is exposed to outside, I would also not consider it private API and would be nice to keep backwards compatibility.